MvvmCross/ViewModels/MvxAppStart.cs

```diff
@@ -88,4 +88,31 @@ protected virtual void NavigateToFirstViewModel(object hint)
             }
         }
     }
+
+    public class MvxAppStart<TViewModel, TParameter> : MvxAppStart<TViewModel> where TViewModel : IMvxViewModel<TParameter>
+    {
+        public MvxAppStart(IMvxApplication application, IMvxNavigationService navigationService) : base(application, navigationService)
+        {
+        }
+
+        protected override void NavigateToFirstViewModel(object hint)
+        {
+            TParameter navParam = default;
+
+            hint = Application.GetAppStartHint(hint);
```

Comment:
I'm not sure why this has to go through the Application class. In the base `ApplicationStartup` already calls into that with the hint in it. I'm thinking that should maybe change to be able to customize the hint.

Reply:
Yeh we can just modify Startup to return the updated hint

Reply:
Yeah sure, that works for me.

Reply:
That introduces a breaking change guys... Can we please avoid that?

Reply:
Yes it does and that is what drove my previous proposal, see my comment below.  If we accept breaking changes, we might as well do it for good and enforce the parameter type to `MvxBundle`.